build.gradle

```diff
@@ -43,34 +43,31 @@ task versionFile {
 
 jar.dependsOn versionFile
 
-uploadArchives {
+publishing {
     repositories {
-        mavenDeployer {
-            repository(url: uri('build/repo'))
+        maven {
+            url = uri("${rootProject.projectDir}/maven-repo")
         }
     }
 }
 
-pluginBundle {
-  website = 'https://github.com/mgk/s3-plugin/blob/master/README.md'
-  vcsUrl = 'https://github.com/mgk/s3-plugin'
 
-  description = 'S3 upload and download plugin'
+gradlePlugin {
+  website.set('https://github.com/mgk/s3-plugin/blob/master/README.md')
+  vcsUrl.set('https://github.com/mgk/s3-plugin')
 
   plugins {
-    s3Plugin {
+    create("s3Plugin") {
       id = 'com.github.mgk.gradle.s3'
+      implementationClass = 'com.github.mgk.gradle.S3Plugin'
+      description = 'S3 upload and download plugin'
       displayName = 'Gradle S3 plugin'
-      tags = ['aws', 's3']
+      tags.set(['aws', 's3'])
     }
   }
 }
 
-task wrapper(type: Wrapper) {
-    gradleVersion = project.gradleVersion
-}
-
-task checkExample(type: GradleBuild, dependsOn: uploadArchives) {
+task checkExample(type: GradleBuild) {
```

Comment:
@mgk I haven't worked with publishing much, so I wasn't sure how to address this. The `uploadArchives` task has been removed. I think as it is, the `check` task will fail, but I'm not sure exactly how to get it running.